src/content/platform-integration/android/c-interop.md

```diff
@@ -82,45 +91,84 @@ $ flutter create --template=package_ffi native_add
 $ cd native_add
 ```
 
-:::note
-You can exclude platforms from `--platforms` that you don't want
-to build to. However, you need to include the platform of
-the device you are testing on.
-:::
-
 This creates a package with C/C++ sources in `native_add/src`.
 These sources are built by the `hook/build.dart` file.
 
 The FFI library can only bind against C symbols,
 so in C++ these symbols are marked `extern "C"`.
 
+:::note
+If you are using the legacy `plugin_ffi`,
 You should also add attributes to indicate that the
 symbols are referenced from Dart,
 to prevent the linker from discarding the symbols
 during link-time optimization.
 `__attribute__((visibility("default"))) __attribute__((used))`.
+:::
 
 The native code is invoked from dart in `lib/native_add_bindings_generated.dart`.
 
 The bindings are generated with [package:ffigen]({{site.pub-pkg}}/ffigen).
 
-### Legacy FFI plugin
+### Platform library (build hooks) {: #platform-library }
+
+To link against a platform library, use the `DynamicLoadingSystem` in your
+`hook/build.dart`.
+
+For example, to load OpenGL ES (v3):
+
+```dart
+      output.assets.code.add(
+        CodeAsset(
+          package: 'my_package',
+          name: 'libGLES_v3',
+          linkMode: DynamicLoadingSystem(Uri.file('libGLES_v3.so')),
+        ),
+      );
+```
+
+You might need to update the Android manifest
+file of the app or plugin if indicated by
+the documentation.
+
+### First-party support
+
+The process for including native code in source
+code or binary form is the same for an app or
+plugin.
+
+### Open-source third-party support
+
+For open source, build the C/C++ sources in your build hook. Or build the
+sources on GitHub actions and download the binary from github artifacts on
+the build hook.
+
+### Closed-source third-party support
+
+For closed source, build the dynamic library somewhere, and download it in
+your build hook.
+
+## Create an FFI plugin (legacy) {: #create-an-ffi-plugin }
+
+Before your library or program can use the FFI library
+to bind to native code, you must ensure that the
+native code is loaded and its symbols are visible to Dart.
+This page focuses on compiling, packaging,
+and loading Android native code within a Flutter plugin or app.
 
 If you need to access the Flutter Plugin API in Swift/Kotlin,
 or if you need to configure a Google Play services runtime on Android,
 you should use the legacy `plugin_ffi` template:
 
 ```console
-$ flutter create --platforms=android,ios,macos,windows,linux --template=plugin_ffi native_add
+$ flutter create --platforms=android --template=plugin_ffi native_add
 ```
 
 This creates a plugin with C/C++ sources in `native_add/src`,
 but uses platform-specific build files (like `android/build.gradle`)
 to build the native code.
 
-## Other use cases
-
-### Platform library
+### Platform library (legacy FFI plugin)
```

Comment:
Do we need this parenthetical in a header?

I understand that this page shows the new (hooks) vs the old (legacy). But what I think would be much cleaner would be to use tabs in the page rather than a potentially confusing set of "legacy" headers. A tab for build hooks and a tab for legacy.